### PR TITLE
refactor courses

### PR DESCRIPTION
**Pull request by gegorov**

### CODE REVIEW

File: src/app/courses/add-course-page/add-course-page.component.ts [hunk 1]
```diff
@@ -1,8 +1,40 @@
 import { Component } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { ICourse, CourseService } from "../../core/index";
```
Comment (kirillpanov) [inline src/app/courses/add-course-page/add-course-page.component.ts]: add empty line after external imports

File: src/app/courses/add-course-page/add-course-page.component.ts [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
 
 @Component({
     selector: "app-add-course-page",
     templateUrl: "./add-course-page.component.html",
     styleUrls: ["./add-course-page.component.scss"]
 })
-export class AddCoursePageComponent {}
+export class AddCoursePageComponent {
+
```
Comment (kirillpanov) [inline src/app/courses/add-course-page/add-course-page.component.ts]: please remove  empty line

File: src/app/courses/add-course-page/add-course-page.component.ts [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
+    private courseService: CourseService;
+    private router: Router;
+
+    public course: ICourse = {
```
Comment (kirillpanov) [inline src/app/courses/add-course-page/add-course-page.component.ts]: please, add comments to public

File: src/app/courses/add-course-page/add-course-page.component.ts [hunk 4]
```diff
@@ -1,8 +1,40 @@
...
+        id: NaN,
```
Comment (kirillpanov) [inline src/app/courses/add-course-page/add-course-page.component.ts]: better to use "new" string

File: src/app/courses/add-course-page/add-course-page.component.ts [hunk 5]
```diff
@@ -1,8 +1,40 @@
...
+        topRated: false,
+        duration: 0,
+        creationDate: (new Date(Date.now())).toUTCString(),
+        description: "",
+        title: ""
+    };
+
+    constructor(
+        courseService: CourseService,
+        route: ActivatedRoute,
```
Comment (kirillpanov) [inline src/app/courses/add-course-page/add-course-page.component.ts]: you injecting route but not assign

File: src/app/shared/directives/go-back/go-back.directive.spec.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { GoBackDirective } from "./go-back.directive";
+
+describe("GoBackDirective", () => {
+    it("should create an instance", () => {
+        // const directive: GoBackDirective = new GoBackDirective();
```
Comment (kirillpanov) [inline src/app/shared/directives/go-back/go-back.directive.spec.ts]: please remove commented code or fix the test

File: src/app/shared/directives/go-back/go-back.directive.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+export class GoBackDirective {
+
+    private location: Location;
+
+    constructor(location: Location) {
+        this.location = location;
+    }
+
+    @HostListener("click")
+    onClick() {
```
Comment (kirillpanov) [inline src/app/shared/directives/go-back/go-back.directive.ts]: public

File: src/app/courses/add-course-page/add-course-page.component.ts (hunk 4 above)
Comment (gegorov) [reply src/app/courses/add-course-page/add-course-page.component.ts]: image

it is a number on backend.

File: src/app/courses/add-course-page/add-course-page.component.ts (hunk 5 above)
Comment (gegorov) [reply src/app/courses/add-course-page/add-course-page.component.ts]: fixed

File: src/app/shared/directives/go-back/go-back.directive.ts (hunk 7 above)
Comment (gegorov) [reply src/app/shared/directives/go-back/go-back.directive.ts]: fixed

File: src/app/shared/directives/go-back/go-back.directive.spec.ts (hunk 6 above)
Comment (gegorov) [reply src/app/shared/directives/go-back/go-back.directive.spec.ts]: removed

**Pull request merged**